src/Microsoft.Performance.SDK.Tests/ProcessingSourceTests.cs

```diff
@@ -137,7 +137,9 @@ public void ManyDataSourcesWhenSubClassReturnsNullProcessorThenThrows()
 
         [TestMethod]
         [UnitTest]
-        public void WhenTableDiscoveryProvidedUsesDiscovery()
+        [DataRow(true)]  // todo: this datarow is temporary and should be removed once the deprecated
+        [DataRow(false)] // ProcessingSource.ProcessingSource(IProcessingSourceTableProvider) constructor is removed
```

Comment:
Suggestion:

refactor this into 3 separate tests:

1. Assert that when using the ctor with no parameters, the expected default is returned from `CreateTableDiscoverer`
2. Assert that when the obsolete ctor is used, the same instance passed to the ctor is returned from `CreateTableDiscoverer`
3. Assert that the table returned from `CreateTableDiscover` is the one used to discover tables when tables are initialized

Reply:
gah, I was writing back to explain why this was all tested... when I realized that SubDataSource's default constructor doesn't test the base class' base constructor. thanks for pointing this out. I'll do a bigger update on this test as you suggest.

Reply:
ok, updated